core/shared/src/main/scala/zio/Config.scala

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio
+
+import scala.util.control.NoStackTrace
+
+/**
+ * A [[zio.Config]] describes the structure of some configuration data.
+ */
+sealed trait Config[+A] { self =>
+
+  /**
+   * Returns a new config that is the composition of this config and the
+   * specified config.
+   */
+  def ++[B](that: Config[B])(implicit zippable: Zippable[A, B]): Config[zippable.Out] =
+    Config.Zipped[A, B, zippable.Out](self, that, zippable)
+
+  /**
+   * Returns a config whose structure is preferentially described by this
+   * config, but which falls back to the specified config if there is an issue
+   * reading from this config.
+   */
+  def ||[A1 >: A](that: Config[A1]): Config[A1] = Config.Fallback(self, that)
+
+  /**
+   * Adds a description to this configuration.
+   */
+  def ??(label: String): Config[A] = Config.Labelled(self, label)
+
+  /**
+   * Returns a new config whose structure is the same as this one, but which
+   * produces a different Scala value, constructed using the specified function.
+   */
+  def map[B](f: A => B): Config[B] = self.mapOrFail(a => Right(f(a)))
+
+  def mapOrFail[B](f: A => Either[Config.Error, B]): Config[B] = Config.MapOrFail(self, f)
+
+  def optional: Config[Option[A]] = self.map(Some(_)) || Config.succeed(None)
+
+  def orElse[A1 >: A](that: Config[A1]): Config[A1] = self || that
+
+  def sequence: Config[Chunk[A]] = Config.Sequence(self)
+
+  def validate[A1 >: A](f: A1 => Boolean): Config[A1] =
+    self.mapOrFail(a =>
+      if (!f(a)) Left(Config.Error.Generic(Chunk.empty, "Validation of configuration data failed")) else Right(a)
+    )
+
+  def zip[B](that: Config[B])(implicit z: Zippable[A, B]): Config[z.Out] = self ++ that
+}
+object Config {
+  final case class Bool(name: String)                                                                  extends Config[Boolean]
+  final case class Constant[A](value: A)                                                               extends Config[A]
+  final case class Decimal(name: String)                                                               extends Config[BigDecimal]
+  final case class Duration(name: String)                                                              extends Config[java.time.Duration]
```

Comment:
Technically the same, but still, wouldn't it be nice to spell it out as `zio.Duration` (instead of `java.time.Duration`)?

Reply:
Fixed